smartlearner/interfaces/view.py

```diff
@@ -9,11 +9,13 @@ def __init__(self):
         self.updates = OrderedDict()
         self.value = None
         self.last_update = -1
+        self.last_epoch = -1
 
     def view(self, status):
-        if self.last_update != status.current_update:
+        if self.last_update != status.current_update or self.last_epoch != status.current_epoch:
```

Comment:
No `current_update` and `current_epoch` are not updated at the same time. (See the [trainer's loop](https://github.com/SMART-Lab/smartlearner/blob/master/smartlearner/trainer.py#L62)).

The only time that would matter is if we have a trainer that doesn't have any update :P. i think we should keep it like this, better safe than sorry. Also, since it is a `or` statement and it performs a short-circuit evaluation, it is as fast.